### PR TITLE
JBTM-3272 Run the string serialization code test against a specific s…

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3272
Adds a test for the code added by commit https://issues.redhat.com/browse/JBTM-3262

MAIN !RTS !AS_TESTS !QA_JTA !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !QA_JTS_JACORB !BLACKTIE !XTS !PERF NO_WIN !TOMCAT !JACOCO !LRA

### CODE REVIEW

File: ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/TestJBTM3262.java [hunk 1]
```diff
@@ -39,15 +43,25 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
 
 /**
  * Verify that a log created by the code prior to the fix for JBTM-3249 can be read by later code
```
Comment (ochaloup) [inline ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/TestJBTM3262.java]: there is referred the wrong JBTM jira here

File: ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/TestJBTM3262.java [hunk 2]
```diff
@@ -39,15 +43,25 @@
...
  */
-public class TestJBTM3249 {
+public class TestJBTM3262 {
```
Comment (ochaloup) [inline ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/TestJBTM3262.java]: Is it really good to name the test based on the Jira name? It's more common to name the testcase base on what functionality it provides.

Comment (mmusgrov) [reply ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/TestJBTM3262.java:51]: A more accurate name would be

```java
read_records_created_before_sha_0b511d470f7ab8dedf585cb14d4900170684621b
```

Anything else is not going to capture the intent of the test - namely to verify that JBTM3262 does not break backwards compatibility.

Other than this is this "PR good to go".

**Comment by mmusgrov** [comment]
@ochaloup Can you take another look at this PR please?

**Pull request merged**